app/src/main/java/org/schabi/newpipe/about/LicenseFragment.kt

```diff
@@ -57,26 +66,73 @@ class LicenseFragment : Fragment() {
             val root: View = componentBinding.root
             root.tag = component
             root.setOnClickListener {
-                activeLicense = component.license
                 compositeDisposable.add(
-                    showLicense(activity, component)
+                    showLicense(component)
                 )
             }
             binding.licensesSoftwareComponents.addView(root)
             registerForContextMenu(root)
         }
-        activeLicense?.let { compositeDisposable.add(showLicense(activity, it)) }
+        activeSoftwareComponent?.let { compositeDisposable.add(showLicense(it)) }
         return binding.root
     }
 
     override fun onSaveInstanceState(savedInstanceState: Bundle) {
         super.onSaveInstanceState(savedInstanceState)
-        activeLicense?.let { savedInstanceState.putSerializable(LICENSE_KEY, it) }
+        activeSoftwareComponent?.let { savedInstanceState.putSerializable(SOFTWARE_COMPONENT_KEY, it) }
+    }
+
+    private fun showLicense(
+        softwareComponent: SoftwareComponent
+    ): Disposable {
+        return if (context == null) {
+            Disposable.empty()
+        } else {
+            val context = requireContext()
+            activeSoftwareComponent = softwareComponent
+            Observable.fromCallable { getFormattedLicense(context, softwareComponent.license) }
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe { formattedLicense ->
+                    val webViewData = Base64.encodeToString(
+                        formattedLicense.toByteArray(), Base64.NO_PADDING
+                    )
+                    val webView = WebView(context)
+                    webView.loadData(webViewData, "text/html; charset=UTF-8", "base64")
+
+                    Localization.assureCorrectAppLanguage(context)
+                    val builder = AlertDialog.Builder(requireContext())
+                        .setTitle(softwareComponent.name)
+                        .setView(webView)
+                        .setOnCancelListener { activeSoftwareComponent = null }
+                        .setOnDismissListener { activeSoftwareComponent = null }
+                    if (softwareComponent == NEWPIPE_SOFTWARE_COMPONENT) {
```

Comment:
I think it's perfectly fine if you show "dismiss" and "open website" even for NewPipe, to further simplify the code and make things consistent. After solving (or dismissing) this comment feel free to merge directly, thanks! The code looks much better

Reply:
I think it is ok to handle NewPipe differently and not show the website button. We already do that in the AboutFragment.
However, we can use the same text on the positive button to close the dialog. I'd not use "ok", but rather "dismiss" or even better just "close".
 @opusforlife2 What button text would you use to close the license dialog? _ok_, _dismiss_ or _close_

Reply:
"done" was not an option 😂

Reply:
Done is also ok